lib/Alchemy/Phrasea/Twig/PhraseanetExtension.php

```diff
@@ -37,9 +37,44 @@ public function getFunctions()
             new \Twig_SimpleFunction('collection_logo', array($this, 'getCollectionLogo'), array(
                 'is_safe' => array('html')
             )),
+            new \Twig_SimpleFunction('record_flags', array($this, 'getRecordFlags'))
         );
     }
 
+    public function getRecordFlags(RecordInterface $record)
+    {
+        $databox =  $this->app['phraseanet.appbox']->get_databox($record->getDataboxId());
+
+        $structure = $this->app['factory.status-structure']->getStructure($databox);
```

Comment:
You should ask for the structure to the record or the databox, `factory.status-structure` is a low-level service.